properties/test_coordinate_transform.py

```diff
@@ -0,0 +1,105 @@
+"""Property tests comparing CoordinateTransformIndex to PandasIndex."""
+
+from collections.abc import Hashable
+from typing import Any
+
+import numpy as np
+import pytest
+
+pytest.importorskip("hypothesis")
+
+import hypothesis.strategies as st
+from hypothesis import given
+
+import xarray as xr
+import xarray.testing.strategies as xrst
+from xarray.core.coordinate_transform import CoordinateTransform
+from xarray.core.indexes import CoordinateTransformIndex
+from xarray.testing import assert_identical
+
+DATA_VAR_NAME = "_test_data_"
+
+
+class IdentityTransform(CoordinateTransform):
+    """Identity transform that returns dimension positions as coordinate labels."""
+
+    def forward(self, dim_positions: dict[str, Any]) -> dict[Hashable, Any]:
+        return {name: dim_positions[name] for name in self.coord_names}
+
+    def reverse(self, coord_labels: dict[Hashable, Any]) -> dict[str, Any]:
+        return {dim: coord_labels[dim] for dim in self.dims}
```

Comment:
not sure if I'm misunderstanding something here, but there appears to be something wrong: why do we pass coordinate names to `dim_positions` (which I think map dimension names to positions), and dimension name into `coord_labels`.

(the example at https://xarray-indexes.readthedocs.io/blocks/transform.html#example-astronomy appears to support my mental model of the coordinate transform)

Reply:
Sorry about that! Fixed now and added the label indexing pieces I really wanted